code/API_definitions/kyc-tenure.yaml

```diff
@@ -0,0 +1,485 @@
+
+    Other errors may be returned as specified below.
+
+    # Identifying a device from the access token
+
+    This specification defines the `phoneNumber` field as optional in API requests, specifically in cases where the API is accessed using a 3-legged access token, and the device can be uniquely identified by the token. This approach simplifies API usage for API consumers by relying on the device information associated with the access token used to invoke the API.
+
+    ## Handling of device information:
+
+    ### Optional phoneNumber for 3-legged tokens:
+
+    - When using a 3-legged access token, the device associated with the access token must be considered as the device for the API request. This means that the phoneNumber is not required in the request, and if included it must identify the same device, therefore **it is recommended NOT to include it in these scenarios** to simplify the API usage and avoid additional validations.
+
+    ### Validation mechanism:
+
+    - The server will extract the device identification from the access token, if available.
+    - If the API request additionally includes a `phoneNumber` parameter when using a 3-legged access token, the API will validate that the device identifier provided matches the one associated with the access token.
+    - If there is a mismatch, the API will respond with a `403 INVALID_TOKEN_CONTEXT` error, indicating that the device information in the request does not match the token.
+
+    ### Error handling for unidentifiable devices:
+
+    - If the `phoneNumber` parameter is not included in the request and the device information cannot be derived from the 3-legged access token, the server will return a `422 UNIDENTIFIABLE_DEVICE` error.
+
+    ### Restrictions for tokens without an associated authenticated identifier:
+
+    - For scenarios which do not have a single device identifier associated to the token during the authentication flow, e.g. 2-legged access tokens, the `phoneNumber` parameter MUST be provided in the API request. This ensures that the device identification is explicit and valid for each API call made with these tokens.
+
+    # Further info and support
+
+    (FAQs will be added in a later version of the documentation)
+
+  version: wip
+  contact:
+    email: sp-kyc@lists.camaraproject.org
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+externalDocs:
+  description: Product documentation at CAMARA
+  url: https://github.com/camaraproject/Tenure
+
+servers:
+  - url: "{apiRoot}/kyc-tenure/vwip"
+    variables:
+      apiRoot:
+        default: https://localhost:443
+        description: API root
+
+tags:
+  - name: Check Subscriber Tenure
+    description: Check details about the length of tenure of the subscriber
+
+paths:
+  /check-tenure:
```

Comment:
In other APIs like Sim swap we have just /check

Reply:
When we would prefer to use `GET /noun`, but need to use a `POST` for privacy reasons, I think `POST /verb-noun` is "more" correct than `POST /verb`. And, indeed, the Commonalities example of `/retrieve-location` supports this, though I agree that they only mandate the verb.

So as part of the "self-documentation" of the API definition, I prefer `POST /check-tenure`, and don't like `POST /check`  being used by any API.